services/purchase_request_copier.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+import argparse
+import os
+
+import knackpy
+
+from config.knack import CONFIG
+import utils
+
+APP_ID = os.getenv("KNACK_APP_ID")
+API_KEY = os.getenv("KNACK_API_KEY")
+
+
+def handle_record_types(knack_record):
+    """
+    Reformat the connection field types and ignore the automatically generated fields
+    """
+    output_record = {}
+    for field_name in knack_record.fields:
+        field_data = knack_record.fields[field_name]
+        # get list of record IDs for connection fields
+        if field_data.field_def.type == "connection":
+            if knack_record.get(field_name):
+                output_record[field_name] = [
+                    item["id"] for item in knack_record.get(field_name)
+                ]
+            else:
+                output_record[field_name] = None
+        # ignoring auto-generated fields
+        elif field_data.field_def.type in [
+            "equation",
+            "concatenation",
+            "auto_increment",
+        ]:
+            continue
+        # All other record types
+        else:
+            output_record[field_name] = knack_record.get(field_name)
+
+    return output_record
+
+
+def assign_requester(data, requester_field_id, copied_by_field_id):
+    """
+    Makes the requester the person who created the copied record.
+    """
+    copied_by = data.pop(copied_by_field_id)
+    data[requester_field_id] = copied_by
+    return data
+
+
+def main(args):
+    # Process arguments
+    app_name = args.app_name
+    container = args.container
+    logger.info(args)
+    config = CONFIG[app_name][container]
+
+    # Use a "free" API call to check for if there's purchase requests to be copied
+    app = knackpy.App(app_id=APP_ID, api_key=None)
+    records = app.get(container, filters=None)
```

Comment:
if this is going to run every couple minutes, it might be worth skipping the metadata download in this step and using the vanilla `knackpy.api` to do the initial record check. only because it's faster and 99.999% of the time there will be no records to process.

but i know this app is small so maybe the metadata download is trivial.

Reply:
another passing thought on this ETL...i wonder if schedule the DAG so that it doesn't run in the wee hours of the night. i guess there's always a chance that Amenity is trying to copy a PR at 3am...

Reply:
Yeah I'm concerned whenever we write these ETLs that will run every minute to not bomb our slack channel. I think this is a use case we can accommodate 😆